### ISSUE TITLE
Add support for per-module enabling/disabling of error codes

### ISSUE THREAD

**Issue by hauntsaninja** [python/mypy#9440]
**Feature**

#9172 implemented global error code disabling/enabling as discussed in #8975. It seems like it was always the intention to consider per-module configuration (https://github.com/python/mypy/issues/8975#issuecomment-641167533), but we don't currently have an open issue for it.

**Pitch**

My personal use case is for tests. Typing tests is often just a chore, but it's nice to have whatever type safety you can get for free. I'd want to be able to do:
```
[mypy]
strict = True

[mypy-tests.*]
allow_untyped_defs = True
allow_untyped_calls = True
disable_error_codes = ["var-annotated"]
```
(For this use case, I think there's actually a fair argument to be made that we should suppress var-annotated aka "needs type annotation" errors when checking untyped defs. If this strikes you as reasonable, I can open another issue. Edit: see #8558 for this)

**Comment by gvanrossum** [comment]
Seems reasonable.

You have a typo, the option is actually called `disable_error_code`. This triggered surprise in me, because at first I thought you were talking about `show_error_codes`. Is it too late to rename that flag to `ignore_error_code`? I guess a problem with that is that the inverse would be `dont-ignore-error-code`, which is kind of awkward.

**Comment by mcous** [comment]
Second this request. For me, I'd really like to disable `func-returns-value` in my tests but keep it active in source

**Comment by JukkaL** [comment]
This sounds like a useful feature. Another potential use case would be migrating legacy code -- you might initially ignore some error codes in code that wasn't originally written with type checking in mind, but you'd want to enable all error codes in new files.

**Comment by chadrik** [comment]
>  Another potential use case would be migrating legacy code 

Exactly this.  I wrote a tool called [mypy-runner](https://pypi.org/project/mypy-runner/) for filtering errors using error codes (first using my own error codes, then the official ones once that feature arrived) as well regexes.  I'd love to not have to use my tool at all!  One of the big downsides of the runner-based approach is that there's a major performance penalty:  dmypy rechecks files with errors (regardless of modified timestamps), and from mypy's perspective all those files still have errors, so reruns of dmypy on our very large codebase are slow.

**Comment by nipunn1313** [comment]
Just ran into a use case for this! I'm migrating our (large) codebase to mypy-protobuf 2.0 - and it requires some backward incompatible changes to enums in pyi stubs. Ideally, I'm able to do this in many smaller diffs.

However, if a migrated module tries to import from a non-migrated module, it hits a `name-defined` error.

Ideally I can disable that error specifically - and reenable it later when the non-migrated module gets migrated.

**Comment by levsa** [comment]
This would be useful to turn off override errors when using fixtures in tests, like:

```python
class A:
     @pytest.fixture
     def b():
        pass

     @pytest.fixture
     def a(b):
          pass
class B(A):
     @pytest.fixture
     def c():
        pass

     @pytest.fixture
     def a(b, c):
          pass
```
